tests/test_crawl.py

```diff
@@ -1,8 +1,12 @@
 import json
 import socket
+import logging
+
+from testfixtures import LogCapture
```

Comment:
There is now a `assertLogs` helper in unittest module: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLogs We've backported it to Django's `django.test.TestCase` (not merged yet) to use in earlier Python versions. Perhaps instead of adding testfixtures dependency, Scrapy can backport it too. That way, Scrapy users can test(of course they will need to update their tests to use e.g. scrapy.test.TestCase instead of unittest.TestCase) their programs output without rely on testfixtures or another dependency.